hack/update-mocks.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates mock files using mockgen.
+# Usage: `hack/update-mocks.sh`.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KARMADA_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+trap EXIT
+
+find_files() {
+  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './.git' \
+        -o -wholename './_output' \
+        -o -wholename './_gopath' \
+        -o -wholename './release' \
+        -o -wholename './target' \
+        -o -wholename '*/third_party/*' \
+        -o -wholename '*/vendor/*' \
+        -o -wholename './staging/src/k8s.io/client-go/*vendor/*' \
+        -o -wholename '*/bindata.go' \
+      \) -prune \
+    \) -name '*.go'
+}
+
+cd "${KARMADA_ROOT}"
+
+echo 'executing go generate command on below files'
+for IFILE in $(find_files | xargs grep --files-with-matches -e '//go:generate mockgen'); do
+    go generate -v "$IFILE"
```

Comment:
![image](https://user-images.githubusercontent.com/17962021/196842235-4b8e36fd-542e-4459-87b7-2174df79ba29.png)


Can we check the `mockgen` command in this script?

Reply:
Sure, it's better idea.

Reply:
Fix, try again